src/osdatahub/NGD/ngd_api.py

```diff
@@ -0,0 +1,212 @@
+              cql_filter: str = None,
+              filter_crs: Union[str, int] = None,
+              max_results: int = 100,
+              offset: int = 0) -> FeatureCollection:
+        """
+        Retrieves features from a Collection
+
+        Args:
+            extent (Extent, optional): An extent, either from a bounding box or a Polygon. Only features within this
+                extent will be returned
+                Available CRS values are: EPSG:27700, EPSG:4326, EPSG:3857, and CRS84. Defaults to CRS84
+            crs (str|int, optional): The CRS for the returned features, either in the format "epsg:xxxx" or an epsg
+                number. e.g. British National Grid can be supplied as "epsg:27700" or 27700.
+                Available CRS values are: EPSG:27700, EPSG:4326, EPSG:7405, EPSG:3857, and CRS84. Defaults to CRS84
+            start_datetime (datetime, optional): Selects features that have a temporal property after the given start
+                time
+            end_datetime (datetime, optional): Selects features that have a temporal property before the given end
+                time
+            cql_filter (str, optional): A filter query in the CQL format. More information about supported CQL operators
+                can be found at https://osdatahub.os.uk/docs/ofa/technicalSpecification
+            filter_crs (str|int, optional): The CRS for a given CQL query, either in the format "epsg:xxxx" or an epsg
+                number. e.g. British National Grid can be supplied as "epsg:27700" or 27700
+                Available CRS values are: EPSG:27700, EPSG:4326, EPSG:7405, EPSG:3857, and CRS84. Defaults to CRS84
+            max_results (int, optional): The maximum number of features to return. Defaults to 100
+            offset (int, optional): The offset number skips past the specified number of features in the collection.
+                Defaults to 0
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+
+        assert check_argument_types()
+        assert max_results > 0, f"Argument max_results must be greater than 0 but was {max_results}"
+        assert offset >= 0, f"Argument offset must be greater than 0 but was {offset}"
+        params = {}
+
+        # Checking validity and preformatting arguments
+        if crs:
+            params["crs"] = get_crs(crs=crs)
+
+        if start_datetime or end_datetime:
+            if start_datetime and end_datetime and start_datetime > end_datetime:
+                raise ValueError("Start time must be before end time")
+
+            start_datetime = start_datetime.isoformat() + "Z" if start_datetime else ".."
+            end_datetime = end_datetime.isoformat() + "Z" if end_datetime else ".."
+
+            params["datetime"] = f"{start_datetime}/{end_datetime}"
+
+        if extent:
+            # If extent is a bounding box, pass it as a bbox parameter
+            if extent.is_bbox:
```

Comment:
Is there an efficiency gain to querying bbox instead of polygon?
- If there isn't, could we skip this bit and negate the is_bbox parameter?
- If there is, is there a more robust way to check it's a bbox than declaring it in the from_bbox constructor, perhaps as a cached property or something?

Reply:
I'm not sure there is an efficiency gain honestly, we might want to test that out. Otherwise we could just implement the spatial filter as a cql filter